src/support.cpp

```diff
@@ -48,6 +48,341 @@ bool AreaSupport::handleSupportModifierMesh(SliceDataStorage& storage, const Set
+                    for (unsigned int upper_part_idx = 0; upper_part_idx < upper_infill_parts.size(); ++upper_part_idx)
+                    {
+                        if (support_infill_part.infill_area.empty())
+                        {
+                            continue;
+                        }
+
+                        // we compute intersection based on support infill areas
+                        const AABB& upper_part_boundary_box = upper_infill_parts[upper_part_idx].outline_boundary_box;
+                        //
+                        // Here we are comparing the **outlines** of the infill areas
+                        //
+                        // legend:
+                        //   ^ support roof
+                        //   | support wall
+                        //   # dense support
+                        //   + less dense support
+                        //
+                        // comparing infill        comparing with outline (this is our approach)
+                        //    ^^^^^^               ^^^^^^
+                        //    ####||^^             ####||^^
+                        //    ######||^^           ######||^^
+                        //    ++++####||           ++++++##||
+                        //    ++++++####           ++++++++##
+                        //
+                        if (upper_part_boundary_box.hit(this_part_boundary_box))
+                        {
+                            relevant_upper_polygons.add(upper_infill_parts[upper_part_idx].infill_area);
+                        }
+                    }
+
+                    less_dense_support = less_dense_support.intersection(relevant_upper_polygons);
+                }
+                if (less_dense_support.size() == 0)
+                {
+                    break;
+                }
+
+                // add new infill_areas_per_combine_per_density for the current density
+                support_infill_part.infill_areas_per_combine_per_density.emplace_back();
+                std::vector<Polygons>& support_area_current_density = support_infill_part.infill_areas_per_combine_per_density.back();
+                const Polygons more_dense_support = support_infill_part.infill_area.difference(less_dense_support);
+                support_area_current_density.push_back(more_dense_support);
+            }
+
+            support_infill_part.infill_areas_per_combine_per_density.emplace_back();
+            std::vector<Polygons>& support_area_current_density = support_infill_part.infill_areas_per_combine_per_density.back();
+            support_area_current_density.push_back(support_infill_part.infill_area);
+
+            assert(support_infill_part.infill_areas_per_combine_per_density.size() != 0 && "support_infill_part.infill_areas_per_combine_per_density should now be initialized");
+#ifdef DEBUG
+            for (auto part_i = 0; part_i < support_infill_part.infill_areas_per_combine_per_density.size(); ++part_i)
```

Comment:
Please don't use `auto` if the type is easy.
I think the type of `part_i` here is determined by `0`, while that constant is not clearly signed or unsigned.

Please change this to `unsigned int`